### ISSUE TITLE
Feature request: add support for babel-plugin-module-resolver 

### ISSUE THREAD

**Issue by atav32** [sverweij/dependency-cruiser#292]
### Context
[babel-plugin-module-resolver](https://github.com/tleunen/babel-plugin-module-resolver) is an alternate way to alias module paths. The syntax is a bit simpler than Webpack's `alias` option.

### Expected Behavior
Would be awesome if `dependency-cruiser` could follow these aliases import paths.

### Current Behavior
Currently, any import path aliased with `babel-plugin-module-resolver` is unresolvable.

### Possible Solution
The module path mapping is pretty straight forward in the Babel config – it's just alias name to relative path.

### Considered alternatives
Could move our module aliases from our Babel config to our Webpack config

**Comment by sverweij** [comment]
Hi @atav32 thanks for raising this issue. 

Babel aliases (as well those in use in parcel and rollup) have been on the [backlog](https://github.com/sverweij/dependency-cruiser/projects/1#card-31008898) for some time, but other things got priority first. I've moved it up a bit. Integrating it into dependency-cruiser will be an interesting challenge (even given the [module resolver api](https://github.com/tleunen/babel-plugin-module-resolver/blob/master/DOCS.md#for-plugin-authors)) - I have a few ideas on how to approach it.

Out of curiosity: What is the reason to use babel-plugin-module-resolver over options e.g. bundlers offer for aliases? 

>It seems babel-plugin-module-resolver is moving into legacy land - a.o. because of the advent of alternatives
> like aliases in tsconfig (typescript), and options bundlers (like webpack) offer:
> - there seems to be no active maintainer (a.o. because the maintainers themselves don't use it anymore ([tleunen](https://github.com/tleunen/babel-plugin-module-resolver/issues/346#issue-399827102) and [fatfisz](https://github.com/tleunen/babel-plugin-module-resolver/issues/346#issuecomment-455350839)) and there's a.t.m. nobody willing to jump in)
> - as a consequence there's quite some open [issues](https://github.com/tleunen/babel-plugin-module-resolver/issues?page=1&q=is%3Aissue+is%3Aopen) (not all bugs b.t.w. but a lot of them can be closed by simple triaging, which doesn't seem to happen)
> - some projects mentioned in the [readme](https://github.com/tleunen/babel-plugin-module-resolver#who-is-using-babel-plugin-module-resolver-) moved on as well e.g. [algolia instantsearch](https://github.com/algolia/instantsearch.js/search?q=babel-plugin-module-resolver&type=Code,) and [next.js](https://github.com/zeit/next.js/commit/e80bbe4b80c648156d9582ae709e381cec733828) don't use it anymore.

**Comment by tpucci** [comment]
I personally use `babel-plugin-module-resolver` with React Native. It works great !

**Comment by atav32** [comment]
@sverweij thanks for pointing out that `babel-plugin-module-resolver` isn't the most active project. There isn't an overriding reason that we're using `babel-plugin-module-resolver`. Just a legacy decision that our team inherited.

I'll take a closer look at Webpack's `alias` option.

**Comment by sverweij** [comment]
hi @atav32 & @tpucci thanks again for bringing this to my attention. 

TL;DR The next version of dependency-cruiser will have babel support (including the plugin/ preset ecosystem) - also see the linked PR (#304 ☝️ ).

=> I'd love to hear any feed back on how it works for you. I've published with the beta tag on npm as `dependency-cruiser@9.4.0-beta-1`

### rationale

@tpucci's remark made me realise there's use cases for having a babel plugin taking care of path aliasing. There might be more than these, but they're already enough to warrant support in dependency-cruiser:

- You might not have a bundler in your tool chain when you're not developing for the web but for e.g. electron or atom (and react native, I guess?). 
- You might not be in a position to use TypeScript (yet or at all)

Next to that: dependency-cruiser uses acorn, and acorn only implements TC39 stage 4. If you're using babel you likely do this because you want to use stuff from earlier stages, like `export * as flakes from 'snowdash'`. In order to enable you to use dependency-cruiser on that as well it'll have to grow babel support too.

**Comment by sverweij** [comment]
From the emoticons on the previous comment I infer it works well for you? 
<img width="220" alt="image" src="https://user-images.githubusercontent.com/4822597/83327451-3c53f100-a27c-11ea-90c8-6110330aa134.png">

I've added some tweaks to make it easier to use and configure and published it as `dependency-cruiser@9.4.0-beta-1` - likely it'll be part of the next release (which follows in the coming days).

**Comment by atav32** [comment]
Oh, has this been removed? I no longer see the documentation for `babelConfig`. And I now get an error with my `.dependency-cruiser.js`:
```
ERROR: The rules file is not valid: data.options should NOT have additional properties.
```

Inside my config, I have
```
  ...
  options: {
    babelConfig: {
      fileName: './babel.config.js',
    },
    ...
  },
  ...
```

**Comment by sverweij** [comment]
@atav32 it should still work. If it doesn't it's a bug. I've tested it manually just now, and it seems to work fine still (both in 9.24.0 and  9.25.1).

=> Could you share our .dependency-cruiser.js?

I've split documentation of rules and options a while ago, which might explain why you couldn't find it; it's now here: [babelConfig - use a babel configuration file](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/options-reference.md#babelconfig-use-a-babel-configuration-file).
